frontend/src/views/MockExams/MockExamsQuestion.tsx

```diff
@@ -0,0 +1,309 @@
+import styled from '@emotion/styled';
+import {
+  Button,
+  Checkbox,
+  FormControlLabel,
+  LinearProgress,
+  Paper,
+  Typography,
+} from '@mui/material';
+import { MockExamQuestionDto } from '@osk/shared';
+import { Flex } from 'reflexbox';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { theme } from '../../theme';
+import { useQuestionState } from '../../hooks/useQuestionState/useQuestionState';
+
+interface MockExamQuestionInterface {
+  question: MockExamQuestionDto;
+  onQuestionSubmit: (questionId: number, answerId: number | undefined) => void;
+  isLastQuestion: boolean;
+}
+
+interface MediaInterface {
+  mediaURL: string | null;
+  hasVideo: boolean | undefined;
+  onVideoEnd: () => void;
+}
+
+export const MockExamsQuestion = ({
+  question,
+  onQuestionSubmit,
+  isLastQuestion,
+}: MockExamQuestionInterface) => {
+  const { answers } = question;
+  const selectedAnswerRef = useRef<number | undefined>(undefined);
+  const [selectedAnswer, setSelectedAnswerState] = useState<number>();
+  const [progress, setProgress] = useState(0);
+  const [state, startReadingQuestion, startAnsweringQuestion, startVideo] =
+    useQuestionState();
+  const [elementaryQuestionsAmount, setElementaryQuestionsAmount] = useState(0);
+  const [advancedQuestionsAmount, setAdvancedQuestionsAmount] = useState(0);
+
+  const setSelectedAnswer = (value: number | undefined) => {
+    setSelectedAnswerState(value);
+    selectedAnswerRef.current = value;
+  };
+  const maxTimeInMs = useMemo(() => {
+    const maxTime =
+      state.type === 'reading'
+        ? question.type.timeToReadTheQuestion
+        : question.type.timeToAnswer;
+    return maxTime * 1000;
+  }, [
+    question.type.timeToAnswer,
+    question.type.timeToReadTheQuestion,
+    state.type,
+  ]);
+
+  const hasMedia = useMemo(() => question.mediaReference !== null, [question]);
+  const hasVideo = useMemo(() => {
+    return hasMedia && question.mediaReference?.endsWith('.wmv');
+  }, [question.mediaReference, hasMedia]);
+
+  const submitQuestion = useCallback(() => {
+    onQuestionSubmit(question.id, selectedAnswerRef.current);
+  }, [onQuestionSubmit, question.id]);
+
+  const nextState = useCallback(() => {
```

Comment:
Let's change up the name of this function. Ideally a function should have a verb in the name. Maybe `goToNextState`?